src/Features/Core/Portable/CodeRefactorings/ConvertNumericLiteral/AbstractConvertNumericLiteralCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,151 @@
+                : NumericKind.Unknown;
+
+            if (kind == NumericKind.Unknown)
+            {
+                return;
+            }
+
+            const string NumericExponent = "e";
+            if (kind == NumericKind.Decimal && numericText.IndexOf(NumericExponent, StringComparison.OrdinalIgnoreCase) > -1)
+            {
+                return;
+            }
+
+            if (kind != NumericKind.Decimal)
+            {
+                RegisterRefactoringWithResult(value.ToString(),
+                    context, root, numericToken, value, suffix, FeaturesResources.Convert_to_decimal);
+            }
+
+            if (kind != NumericKind.Binary)
+            {
+                RegisterRefactoringWithResult(binaryPrefix + Convert.ToString(value, 2),
+                    context, root, numericToken, value, suffix, FeaturesResources.Convert_to_binary);
+            }
+
+            if (kind != NumericKind.Hexadecimal)
+            {
+                RegisterRefactoringWithResult(hexPrefix + value.ToString("X"),
+                    context, root, numericToken, value, suffix, FeaturesResources.Convert_to_hex);
+            }
+
+            const string DigitSeparator = "_";
+            if (numericText.Contains(DigitSeparator))
+            {
+                RegisterRefactoringWithResult(numericText.Replace(DigitSeparator, string.Empty),
+                    context, root, numericToken, value, string.Empty, FeaturesResources.Remove_separators);
+            }
+            else
+            {
+                switch (kind)
+                {
+                    case NumericKind.Decimal:
+                        RegisterRefactoringWithResult(AddSeparators(number, 3),
+                            context, root, numericToken, value, suffix, FeaturesResources.Separate_thousands);
+                        break;
+
+                    case NumericKind.Hexadecimal:
+                        RegisterRefactoringWithResult(hexPrefix + AddSeparators(number, 4),
+                            context, root, numericToken, value, suffix, FeaturesResources.Seperate_words);
+                        break;
+
+                    case NumericKind.Binary:
+                        RegisterRefactoringWithResult(binaryPrefix + AddSeparators(number, 4),
+                            context, root, numericToken, value, suffix, FeaturesResources.Seperate_nibbles);
+                        break;
+                }
+            }
+        }
+
+        private static void RegisterRefactoringWithResult(
+            string numericText, CodeRefactoringContext context, SyntaxNode root, SyntaxToken numericToken, long value, string suffix, string title)
+            => context.RegisterRefactoring(new MyCodeAction(title, c => Task.FromResult(context.Document.WithSyntaxRoot(
+                root.ReplaceToken(numericToken, SyntaxGenerator.GetGenerator(context.Document).NumericLiteralToken(numericText + suffix, value))))));
```

Comment:
Sure, but I still think this should be a local function. It just doesn't feel right otherwise. There should be *some* acceptable usages if that made the cut for C# 7. :/